test/GoodGhostingPolygon.test.js

```diff
@@ -904,6 +904,212 @@ contract("GoodGhostingPolygon", (accounts) => {
         });
     });
 
+    describe("when reward and deposit token or incentive and deposit token or incentive and reward token are same", async () => {
+        beforeEach(async () => {
+            pap = await LendingPoolAddressesProviderMock.new("TOKEN_NAME", "TOKEN_SYMBOL", { from: admin });
+            aToken = await IERC20.at(await pap.getLendingPool.call());
+            await pap.setUnderlyingAssetAddress(incentiveController.address);
+            await incentiveController.mint(player1, toWad(1000), { from: admin });
+            await incentiveController.mint(player2, toWad(1000), { from: admin });
+            contract = await GoodGhostingPolygon.new(
+                incentiveController.address,
+                pap.address,
+                segmentCount,
+                segmentLength,
+                segmentPayment,
+                fee,
+                0,
+                pap.address,
+                "115792089237316195423570985008687907853269984665640564039457584007913129639935", // equals to 2**256-1
+                ZERO_ADDRESS,
+                incentiveController.address,
+                incentiveController.address,
+                { from: admin },
+            );
+        });
```

Comment:
This before each is only used for the first 3 `it` blocks below. For the forth and on, they have their own deployment logic. We could wrap those 3 `it` tests and the `beforeEach` block inside it's own `describe` - it can continue to be contained by the current outer `describe`. just a small improvement to try to decrease the total testing time